src/com/massivecraft/factions/engine/EnginePermBuild.java

```diff
@@ -209,6 +211,30 @@ public void buildEntity(EntityDamageByEntityEvent event)
 		build(player, entity.getLocation().getBlock(), event);
 	}
 	
+	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+	public void hitEntity(ProjectileHitEvent event)
+	{
+		// If a player's projectile ...
+		ProjectileSource projectileSource = event.getEntity().getShooter();
+		if (!(projectileSource instanceof Entity) || MUtil.isntPlayer(projectileSource)) return;
```

Comment:
It seems redundant to have both of these checks. If the projectileSource is a player, then it follows by necessity that the projectileSource is also an entity. It should be sufficient to use `if (MUtil.isntPlayer(projectileSource)) return;`

Reply:
Agree  - Must have been a bit tired writing that line. 
Updated PR reflects changes.